pbsmetrics/prometheus/prometheus.go

```diff
@@ -1,585 +1,373 @@
-	labels = addDimension(labels, requestTypeLabel, requestTypesAsString())
-	labels = addDimension(labels, browserLabel, browserTypesAsString())
-	labels = addDimension(labels, cookieLabel, cookieTypesAsString())
-	adapterLabels := labels // save regenerating these dimensions for adapter status
-	labels = addDimension(labels, responseStatusLabel, requestStatusesAsString())
-	// If we implement an account whitelist, we can seed the metrics with that list to redusce latency associated with registering new lable values on the fly.
-	labels = addDimension(labels, accountLabel, []string{pbsmetrics.PublisherUnknown})
-	for _, l := range labels {
-		_ = m.requests.With(l)
-		_ = m.reqTimer.With(l)
-	}
-
-	// Adapter labels
-	labels = addDimension(adapterLabels, adapterLabel, adaptersAsString())
-	errorLabels := labels // save regenerating these dimensions for adapter errors
-	labels = addDimension(labels, adapterBidLabel, adapterBidsAsString())
-	for _, l := range labels {
-		_ = m.adaptRequests.With(l)
-		_ = m.adaptTimer.With(l)
-		_ = m.adaptPrices.With(l)
-		_ = m.adaptPanics.With(l)
-	}
-
-	// AdapterBid labels
-	labels = addDimension(labels, bidTypeLabel, bidTypesAsString())
-	labels = addDimension(labels, markupTypeLabel, []string{"unknown", "adm"})
-	for _, l := range labels {
-		_ = m.adaptBids.With(l)
-	}
-	labels = addDimension(errorLabels, adapterErrLabel, adapterErrorsAsString())
-	for _, l := range labels {
-		_ = m.adaptErrors.With(l)
-	}
-	cookieLabels := addDimension([]prometheus.Labels{}, adapterLabel, adaptersAsString())
-	cookieLabels = addDimension(cookieLabels, gdprBlockedLabel, []string{"true", "false"})
-	for _, l := range cookieLabels {
-		_ = m.adaptCookieSync.With(l)
-	}
-	cacheLabels := addDimension([]prometheus.Labels{}, "cache_result", cacheResultAsString())
-	for _, l := range cacheLabels {
-		_ = m.storedImpCacheResult.With(l)
-		_ = m.storedReqCacheResult.With(l)
-	}
-
-	// ImpType labels
-	impTypeLabels := addDimension([]prometheus.Labels{}, bannerLabel, []string{"yes", "no"})
-	impTypeLabels = addDimension(impTypeLabels, videoLabel, []string{"yes", "no"})
-	impTypeLabels = addDimension(impTypeLabels, audioLabel, []string{"yes", "no"})
-	impTypeLabels = addDimension(impTypeLabels, nativeLabel, []string{"yes", "no"})
-	for _, l := range impTypeLabels {
-		_ = m.imps.With(l)
-	}
-}
-
-// addDimesion will expand a slice of labels to add the dimension of a new set of values for a new label name
-func addDimension(labels []prometheus.Labels, field string, values []string) []prometheus.Labels {
-	if len(labels) == 0 {
-		// We are starting a new slice of labels, so we can't loop.
-		return addToLabel(make(prometheus.Labels), field, values)
-	}
-	newLabels := make([]prometheus.Labels, 0, len(labels)*len(values))
-	for _, l := range labels {
-		newLabels = append(newLabels, addToLabel(l, field, values)...)
+func (m *Metrics) RecordAdapterBidReceived(labels pbsmetrics.AdapterLabels, bidType openrtb_ext.BidType, hasAdm bool) {
```

Comment:
Although we're not using it as a dimension for Prometheus, it is being used by the Influx metrics. The method signature must include it to satisfy the logging interface. We can follow-up with perhaps an update of the Influx metrics to better align with the decisions we made in here. .. or, do you think the adm vs nurl metric is more useful per bid type? There's an extra cost because it's per-adapter, but not crazy so.